### ISSUE TITLE
Logging is broken with the switch to klog/v2

### ISSUE THREAD

**Issue by damemi** [kubernetes-sigs/descheduler#353]
Testing a build with and without the changes in https://github.com/kubernetes-sigs/descheduler/pull/327, the switch to klog/v2 seems to have broken logging and nothing is printed, even at log level 5.

**What version of descheduler are you using?**
descheduler version: built from latest master

**Does this issue reproduce with the latest release?**
not 1.18, yes for 1.19

**Which descheduler CLI options are you using?**
```
        - --policy-config-file=/policy-dir/policy.yaml
        - --v=5
        - --descheduling-interval=3600s
```

**Please provide a copy of your descheduler policy config file**
```
  policy.yaml: |
    strategies:
      LowNodeUtilization:
        enabled: true
        params:
          nodeResourceUtilizationThresholds:
            numberOfNodes: 3
            targetThresholds:
              cpu: 10
              memory: 20
              pods: 30
            thresholds:
              cpu: 10
              memory: 20
              pods: 30
      RemoveDuplicates:
        enabled: true
```

**What k8s version are you using (`kubectl version`)?**
```
$ oc version
Client Version: v4.2.0-alpha.0-361-gcb858ac
Server Version: 4.6.0-0.nightly-2020-07-15-170241
Kubernetes Version: v1.18.3+7024740
```

/priority critical-urgent

**Comment by damemi** [comment]
/assign @farah 
Can you please take a look at this? So far I've tried adding `klog.InitFlags(nil)` to cmd/descheduler/descheduler.go but that causes an error: 
```
/bin/descheduler flag redefined: log_dir
panic: /bin/descheduler flag redefined: log_dir

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc000184120, 0x1736320, 0x2140f90, 0x152efe5, 0x7, 0x155c0fd, 0x2f)
	/usr/local/go/src/flag/flag.go:851 +0x4b8
flag.(*FlagSet).StringVar(...)
	/usr/local/go/src/flag/flag.go:754
k8s.io/klog/v2.InitFlags(0x0)
	/go/src/sigs.k8s.io/descheduler/vendor/k8s.io/klog/v2/klog.go:425 +0x7f
main.main()
	/go/src/sigs.k8s.io/descheduler/cmd/descheduler/descheduler.go:32 +0x95
```

indicating to me that there is some conflict with initializing the v2 flags. if there isn't a quick fix I would recommend reverting the changes in #327 for the 1.19 release

/cc @seanmalloy @ingvagabund

**Comment by farah** [comment]
@damemi How long do I have? It's passed midnight here so will try find a fix in the morning.

I'll refer to this [demo](https://github.com/kubernetes/klog/pull/142/files)

**Comment by farah** [comment]
@damemi How long do I have? It's passed midnight here so will try find a fix in the morning.

**Comment by damemi** [comment]
@farah I'll keep working on this today to try to find a fix. I've also looked at the [klog/glog coexist example](https://github.com/kubernetes/klog/blob/b5c3182dac44f851522e32c97c86ac32755c296d/examples/coexist_glog/coexist_glog.go).

Doesn't need to be fixed immediately but we do need it soon, so any help is appreciated. Thanks!

**Comment by damemi** [comment]
Update: this *may* be fixed with the bump of our upstream dependencies to 1.19 (see https://github.com/kubernetes-sigs/descheduler/pull/337), but I haven't been able to verify it yet because I hit [this "unknown conversion" error](https://github.com/openshift/cluster-kube-descheduler-operator/issues/116) trying to decode our policies.

Like I mention in that issue, that error could just be due to OpenShift's operator. However it would be very helpful if someone has the time to try building and running a descheduler from my 1.19 branch in #337 to see if 1) that fixes the logging issue and 2) verify whether we have an issue with go 1.14

**Comment by damemi** [comment]
Verified that the "unknown conversion" error is only present in my 1.19 rebase branch, **not** master. So that error isn't specific to OpenShift, but it is a 1.19 problem I'll just need to solve in #337. So disregard that error.

However, I still have not been able to figure out how to re-enable logging with klog v2. I do see some throttling log messages:
```
$ oc logs pod/descheduler-job-nf7hz
I0717 17:16:56.357223       1 request.go:557] Throttling request took 146.416983ms, request: GET:https://172.30.0.1:443/api/v1/pods?fieldSelector=spec.nodeName%3Dci-ln-ythq5gb-f76d1-dxddq-worker-d-mpj77%2Cstatus.phase%21%3DFailed%2Cstatus.phase%21%3DSucceeded
I0717 17:16:56.557213       1 request.go:557] Throttling request took 134.431767ms, request: GET:https://172.30.0.1:443/api/v1/pods?fieldSelector=spec.nodeName%3Dci-ln-ythq5gb-f76d1-dxddq-master-0%2Cstatus.phase%21%3DFailed%2Cstatus.phase%21%3DSucceeded
I0717 17:16:56.757165       1 request.go:557] Throttling request took 149.147355ms, request: GET:https://172.30.0.1:443/api/v1/pods?fieldSelector=spec.nodeName%3Dci-ln-ythq5gb-f76d1-dxddq-master-1%2Cstatus.phase%21%3DFailed%2Cstatus.phase%21%3DSucceeded
```

but nothing else. Makes me wonder if the default for something like `logtostderr` changed to false

**Comment by ingvagabund** [comment]
Descheduler is based on cobra which is implementing its own `flag` library. Whereas, klog/v2 is using go flag package. So there are actually two flagSets initialized. One for klog/v1 and the other one for klog/v2. Though, only the klog/v1 flag set is actually used.

**Comment by damemi** [comment]
@farah @ingvagabund good news! I was able to get the [1.19 rebase](https://github.com/kubernetes-sigs/descheduler/pull/337) to compile and run and it looks like logging *does* work with upstream deps bumped to 1.19.

I noticed that some deps, like `component-base`, that provide logging flags made the switch to `klog/v2` in 1.19. I think there must have been some conflict in how the flags/logging libraries were initialized that wasn't captured while we have the version skew.

That means that (hopefully) #337 should fix this. Thanks for reacting quickly and offering your help, everyone!
